makefile

```diff
@@ -5,54 +5,57 @@ ifeq ($(WORKSPACE),)
 	WORKSPACE := $(shell pwd)/../../../../
 endif
 
+export GO_LDFLAGS=-ldflags "-s -w -X github.com/aws/aws-xray-daemon/pkg/cfg.Version=${VERSION}"
+
 # Initialize BGO_SPACE
 export BGO_SPACE=$(shell pwd)
 path := $(BGO_SPACE):$(WORKSPACE)
 
-build: pre-build create-folder copy-file build-mac build-linux build-windows zip-linux zip-osx zip-win
+build: create-folder copy-file build-mac build-linux build-linux-arm64 build-windows
 
-packaging: package-rpm package-deb
+packaging: zip-linux zip-osx zip-win package-rpm package-deb
 
 release: build test packaging clean-folder
 
-.PHONY: pre-build
-pre-build:
-	go run $(BGO_SPACE)/Tool/src/versiongenerator/version-gen.go
-
 .PHONY: create-folder
 create-folder:
 	mkdir -p build/xray
+	mkdir -p build/dist
 
 .PHONY: copy-file
 copy-file:
 	cp pkg/cfg.yaml build/xray/
-	cp $(BGO_SPACE)/VERSION	build/xray/
 	cp $(BGO_SPACE)/LICENSE build/xray
 	cp $(BGO_SPACE)/THIRD-PARTY-LICENSES.txt build/xray
 
 .PHONY: build-mac
 build-mac:
 	@echo "Build for MAC amd64"
-	GOOS=darwin GOARCH=amd64 go build -ldflags "-s -w" -o $(BGO_SPACE)/build/xray/xray_mac ${PREFIX}/cmd/tracing/daemon.go ${PREFIX}/cmd/tracing/tracing.go
+	GOOS=darwin GOARCH=amd64 CGO_ENABLED=0 go build $(GO_LDFLAGS) -o $(BGO_SPACE)/build/xray-mac-amd64/xray ${PREFIX}/cmd/tracing/daemon.go ${PREFIX}/cmd/tracing/tracing.go
 
 .PHONY: build-linux
 build-linux:
 	@echo "Build for Linux amd64"
-	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -ldflags "-s -w" -o $(BGO_SPACE)/build/xray/xray ${PREFIX}/cmd/tracing/daemon.go ${PREFIX}/cmd/tracing/tracing.go
+	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build $(GO_LDFLAGS) -o $(BGO_SPACE)/build/xray-linux-amd64/xray ${PREFIX}/cmd/tracing/daemon.go ${PREFIX}/cmd/tracing/tracing.go
+
+.PHONY: build-linux-arm64
+build-linux-arm64:
+	@echo "Build for Linux arm64"
+	GOOS=linux GOARCH=arm64 CGO_ENABLED=0 go build $(GO_LDFLAGS) -o $(BGO_SPACE)/build/xray-linux-arm64/xray ${PREFIX}/cmd/tracing/daemon.go ${PREFIX}/cmd/tracing/tracing.go
 
 .PHONY: build-windows
 build-windows:
 	@echo "Build for Windows amd64"
-	GOOS=windows GOARCH=amd64 go build -ldflags "-s -w" -o $(BGO_SPACE)/build/xray/xray.exe ${PREFIX}/cmd/tracing/daemon.go ${PREFIX}/cmd/tracing/tracing_windows.go
-	GOOS=windows GOARCH=amd64 go build -ldflags "-s -w" -o $(BGO_SPACE)/build/xray/xray_windows.exe ${PREFIX}/cmd/tracing/daemon.go ${PREFIX}/cmd/tracing/tracing.go
+	GOOS=windows GOARCH=amd64 CGO_ENABLED=0 go build $(GO_LDFLAGS) -o $(BGO_SPACE)/build/xray-windows-amd64/xray_service.exe ${PREFIX}/cmd/tracing/daemon.go ${PREFIX}/cmd/tracing/tracing_windows.go
+	GOOS=windows GOARCH=amd64 CGO_ENABLED=0 go build $(GO_LDFLAGS) -o $(BGO_SPACE)/build/xray-windows-amd64/xray.exe ${PREFIX}/cmd/tracing/daemon.go ${PREFIX}/cmd/tracing/tracing.go
 
 .PHONY: build-docker
```

Comment:
Should we also remove build and push to docker from makefile? Since we have a different job that will push daemon image to ECR and docker hub.

Reply:
Yeah I'm thinking of tweaking the docker build in a follow-up